common/src/main/scala/org/apache/celeborn/common/metrics/source/ResourceConsumptionSource.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.metrics.source
+
+import org.apache.celeborn.common.CelebornConf
+import org.apache.celeborn.common.internal.Logging
+import org.apache.celeborn.common.metrics.MetricsSystem
+
+class ResourceConsumptionSource(conf: CelebornConf)
+  extends AbstractSource(conf, MetricsSystem.ROLE_MASTER) with Logging {
+  override val sourceName = "ResourceConsumption"
```

Comment:
Could we add it into Master source if it will only be used by Master?

Reply:
Prefer to have a divided one since these kinds of metrics is special, we can have a sepecial source name